install/helm/agones/templates/service/allocation.yaml

```diff
@@ -0,0 +1,189 @@
+        secret:
+          secretName: allocator-client-certs
+      containers:
+      - name: agones-allocator
+        image: "{{ .Values.agones.image.registry }}/{{ .Values.agones.image.allocator.name}}:{{ default .Values.agones.image.tag .Values.agones.image.allocator.tag }}"
+        imagePullPolicy: {{ .Values.agones.image.controller.pullPolicy }}
+        ports:
+        - name: https
+          containerPort: 8443
+        volumeMounts:
+        - mountPath: /home/allocator/tls
+          name: tls
+          readOnly: true
+        - mountPath: /home/allocator/client-certs
+          name: client-certs
+          readOnly: true
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+
+---
+# Create a Role in the default namespace that grants access to the agones api
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: gameserver-allocator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "agones.name" $ }}
+    chart: {{ template "agones.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+rules:
+- apiGroups: ["allocation.agones.dev"]
+  resources: ["gameserverallocations"]
+  verbs: ["create"]
+
+---
+# Create a ServiceAccount that will be bound to the above role
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: gameserver-allocator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "agones.name" $ }}
+    chart: {{ template "agones.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+
+---
+# Bind the gameserver-allocator ServiceAccount to the gameserver-allocator Role
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: gameserver-allocator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "agones.name" $ }}
+    chart: {{ template "agones.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+subjects:
+- kind: ServiceAccount
+  name: gameserver-allocator
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: Role
+  name: gameserver-allocator
+ 
+{{- end }}
+
+---
+# Allocation TLS certs
+{{- $ca := genCA "allocation-tls-ca" 3650 }}
+{{- $cert := genSignedCert "" nil nil 3650 $ca }}
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/tls
+metadata:
+  name: allocator-tls
+  labels:
+    app: {{ template "agones.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data:
+{{- if .Values.agones.allocator.generateTLS }}
+  tls.crt: {{ b64enc $cert.Cert }}
+  tls.key: {{ b64enc $cert.Key }}
+{{- else }}
+  tls.crt: {{ .Files.Get "certs/allocator/server.crt" | b64enc }}
+  tls.key: {{ .Files.Get "certs/allocator/server.key" | b64enc }}
+{{- end }}
+
+---
+# Allocation client certs
+{{- $ca := genCA "allocation-client-ca" 3650 }}
```

Comment:
pretty sure you want one CA and not separate CA for client and server

Reply:
Server CA is needed if the server is self-signed. Otherwise, client can confirm the server's validity by consulting with CAs signed by valid certificate authorities e.g. DigiCert. Client CA list indicates a list of clients that a server accepts and acts as an authN.